boards: nrf52840 common uf2 partition

jacobw opened the pull request:
Common uf2 partition configuration to avoid duplication in boards.

There appears to be a bit of confusion about exactly what a proper partition map looks like for the nrf52840, so a common dts configuration should help to avoid confusion.

xudongzheng reviewed (changes requested):
There are essentially three variants of the Adafruit UF2 bootloader:

- No SoftDevice (offset `0x1000`)
- SoftDevice v6 (offset `0x26000`)
- SoftDevice v7 (offset `0x27000`)

It seems like this change is combining both the SoftDevice v6 and v7 partition tables into the v6 partition table. AFAIK most of the Adafruit boards ship with the v6 bootloader whereas the Seeed boards ship with the v7 bootloader.

I would suggest separate files `nrf52840_partition_uf2_sdv6.dtsi` and `nrf52840_partition_uf2_sdv7.dtsi`. Possibly `nrf52840_partition_uf2_nosd.dtsi` as well though I don't think Zephyr has any in-tree boards that use the bootloader without SoftDevice.

jacobw commented:
I've updated based on your comments @xudongzheng. I looks like I'm not able to see your suggested changes after my force push so apologies if I've missed some of your suggestions.

jacobw commented:
ping @anangl

jacobw commented:
Thanks @anangl 
ping @masz-nordic

masz-nordic commented on dts/common/nordic/nrf52840_partition_uf2_sdv7.dtsi:
```diff
@@ -0,0 +1,52 @@
+		reserved_partition_0: partition@0 {
+			label = "SoftDevice";
+			read-only;
+			reg = <0x00000000 0x00026000>;
```
Shouldn't this be `0x00027000`?

jacobw replied:
Yes, looks like I missed that.

jacobw committed:
boards: nrf52840 common uf2 partition

Common uf2 partition configurations to avoid duplication in boards.

There appears to be a bit of confusion about exactly what a proper
UF2 partition map looks like for the nrf52840, so common dts
configurations should help to avoid confusion.

Configuration for SoftDevice v6 and v7 provided as thats what was
fouond in use in tree already.

Signed-off-by: Jacob Winther <jacob@9.nz>
